integration_test/arcilator/JIT/dpi.mlir

```diff
@@ -0,0 +1,39 @@
+// RUN: arcilator %s --run --jit-entry=main | FileCheck %s
+// REQUIRES: arcilator-jit
+
+// CHECK:      c = 0
+// CHECK-NEXT: c = 5
+sim.func.dpi @dpi(in %a : i32, in %b : i32, out c : i32) attributes {verilogName = "adder_func"}
+func.func @adder_func(%arg0: i32, %arg1: i32, %arg2: !llvm.ptr) {
+  %0 = arith.addi %arg0, %arg1 : i32
+  llvm.store %0, %arg2 : i32, !llvm.ptr
+  return
+}
+hw.module @adder(in %clock : i1, in %a : i32, in %b : i32, out c : i32) {
+  %seq_clk = seq.to_clock %clock
+
+  %0 = sim.func.dpi.call @dpi(%a, %b) clock %seq_clk : (i32, i32) -> i32
```

Comment:
Just out of curiosity: since the `sim.func.dpi.call` function seems like it accepts both `sim.func.dpi` _and_ `func.func` callables, would this code also work with a second `sim.func.dpi.call` that calls the `@adder_func` directly?

Reply:
Directly calling `@adder_func` doesn't work since it's output value is passed by reference in argument. If `@adder_func` was a normal dataflow-ish func.func such as `func.func @adder_func_ok(%arg0: i32, %arg1: i32) -> i32`, we can call it directly.

Reply:
Ooooh you're right, sorry, I overlooked that difference in the operand layout. Makes sense!